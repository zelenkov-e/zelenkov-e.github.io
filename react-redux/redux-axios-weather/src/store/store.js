import redusers from "../reducer/index";
import { applyMiddleware,createStore } from "redux";
import { createLogger } from 'redux-logger';
import thunk from 'redux-thunk';//<-- добавили redux-thunk






export default function  configureStore(initialState) {
	const logger = createLogger()
	const store = createStore(
		redusers,
		initialState,
		//добавили в applyMiddleware(createLogger()) 
		applyMiddleware(thunk,logger))// <-- добавили его в цепочку перед logger'ом
	return store

}



