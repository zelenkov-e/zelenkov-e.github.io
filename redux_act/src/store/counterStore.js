import { createStore } from "redux";
import { rootReducer } from "../reducer/combineReducer";
import { applyMiddleware } from "redux";
import { createLogger } from "redux-logger";
import thunk from "redux-thunk"; //<-- добавили redux-thunk

function configureStore(initialState) {
  const logger = createLogger();
  const store = createStore(
    rootReducer,
    initialState,
    //добавили в applyMiddleware(createLogger())
    applyMiddleware(thunk, logger)
  ); // <-- добавили его в цепочку перед logger'ом
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();
  return store;
}

export const counterStore = configureStore();

// export const counterStore = createStore(
//   rootReducer,
//   window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
// );
