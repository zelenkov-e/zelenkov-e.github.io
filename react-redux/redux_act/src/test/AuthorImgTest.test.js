import React from "react";
import { shallow } from "enzyme";
import AuthorImg from "../js/authorImg";
import { counterStore } from "../store/counterStore";

describe("AuthorImg", () => {
  it(" render my component", () => {
    const wrapper = shallow(<AuthorImg store={counterStore} />);
    expect(wrapper).toMatchSnapshot();
  });
});
