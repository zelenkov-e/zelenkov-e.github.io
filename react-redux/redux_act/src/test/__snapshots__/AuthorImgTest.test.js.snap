// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorImg  render my component 1`] = `
<AuthorImg
  dispatch={[Function]}
  elementAuthor={
    Object {
      "audio": null,
      "author": Array [],
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
